server/helpers_test.go

```diff
@@ -36,3 +37,25 @@ type stubOCSPRespSource struct {
 func (s *stubOCSPRespSource) Fetch(*certchain.RawChain, func() time.Time) (*certchain.OCSPResponse, futureevent.Event, error) {
 	return s.data, futureevent.NeverOccurs(), nil
 }
+
+//------------------------
+//  stubCache
+
+var _ certmanager.Cache = (*stubCache)(nil)
+
+type stubCache struct {
+	OnWrite chan *certchain.AugmentedChain
+}
+
+func newStubCache() *stubCache {
+	return &stubCache{make(chan *certchain.AugmentedChain, 1)}
+}
+
+func (c *stubCache) Read(digest string) (*certchain.AugmentedChain, error) {
+	return <-c.OnWrite, nil
```

Comment:
See my comment above. For this stub, I think you can just define a field to keep the AugmentedChain and get rid of `OnWrite`.

Reply:
Done